elip-0101.mediawiki

```diff
@@ -0,0 +1,100 @@
+
+===Copyright===
+
+This ELIP is licensed under the 2-clause BSD license.
+
+==Specification==
+
+The Partially Signed Elements Transaction (PSET) format extends the BIP 370 PSBT format and is defined at https://github.com/ElementsProject/elements/blob/master/doc/pset.mediawiki. This document extends the PSET format directly, and as such the document above should be updated if and when this ELIP is approved.
+
+The new global field is defined as follows:
+
+{|
+! Name
+! <tt><keytype></tt>
+! <tt><keydata></tt>
+! <tt><keydata></tt> Description
+! <tt><valuedata></tt>
+! <tt><valuedata></tt> Description
+! Versions Requiring Inclusion
+! Versions Requiring Exclusion
+! Versions Allowing Inclusion
+|-
+| Genesis Blockhash
+| <tt>PSBT_ELEMENTS_GLOBAL_GENESIS_HASH = 0x02</tt>
+| None
+| No key data
+| <tt><hash></tt>
+| The 32 byte genesis hash of the Elements chain this transaction is for.
+| 2(*)
+| -
+| 2
+|}
+
+(*) Note that required inclusion is conditional as described below.
+
+===Handling Duplicated Keys===
+
+Duplicate or repeated values of the new field are not allowed. When merging multiple PSETs, if any input PSET has this field, the resulting PSET must also. If more than one value for this field is present or would result, merging must fail. These are the standard rules for existing fields and should not require special handling.
+
+Logically, a PSET must either explicitly belong to a given chain (the field is present), or not explicitly belong to any chain (the field is absent).
+
+==Roles==
+
+For all roles, when a PSET is received and the field is present, it should be validated if context is available. If the genesis hash does not match the current context, the operation must fail.
+
+The field must not ever be removed, even after finalization, since it can be used to prevent further processing or broadcast on the wrong chain.
+
+Otherwise, the PSET Roles are updated as follows:
+
+===Creator/Updater===
+
+May add the field to any PSET they process, if context is available.
+
+If taproot inputs are added, they must add the field.
+
+If the PSET already contains taproot inputs and the field is not present, an updater should add it if context is available.
+
+===Signer===
+
+If present, the field must be validated if context is available.
+
+If not present, signers may sign taproot inputs with their current chain if context is available.
+
+If any inputs of any type are signed the field should be added, and must be added if taproot inputs are present (whether signed or not).
+
+
+<references/>
+
+==Backwards Compatibility==
+
+The new field is backwards compatible due to the ignore/round trip behavior for new fields inherited from the PSBT specification. Conforming implementations are expected to carry through the new field and its contents even if they do not support this ELIP.
+
+==Reference implementation==
+
+TBD in Elements: https://github.com/ElementsProject/elements
```

Comment:
It's worth mentioning wally implementation here https://github.com/ElementsProject/libwally-core/blob/master/CHANGES.md#added

Reply:
done